Mathlib/Analysis/SpecialFunctions/Complex/CircleAddChar.lean

```diff
@@ -31,17 +31,26 @@ open AddCircle
 
 namespace ZMod
 
+/-!
+### Additive characters valued in the complex circle
+-/
+
+open scoped Real
+
 variable {N : ℕ} [NeZero N]
 
 /-- The additive character from `ZMod N` to the unit circle in `ℂ`, sending `j mod N` to
 `exp (2 * π * I * j / N)`. -/
-noncomputable def toCircle : AddChar (ZMod N) circle :=
-  toCircle_addChar.compAddMonoidHom toAddCircle
+noncomputable def toCircle : AddChar (ZMod N) Circle where
+  toFun := fun j ↦ (toAddCircle j).toCircle
+  map_add_eq_mul' a b := by simp_rw [map_add, AddCircle.toCircle_add]
+  map_zero_eq_one' := by simp_rw [map_zero, AddCircle.toCircle, ← QuotientAddGroup.mk_zero,
+    Function.Periodic.lift_coe, mul_zero, Circle.exp_zero]
 
 lemma toCircle_intCast (j : ℤ) :
     toCircle (j : ZMod N) = exp (2 * π * I * j / N) := by
-  rw [toCircle, AddChar.compAddMonoidHom_apply, toCircle_addChar, AddChar.coe_mk,
-    AddCircle.toCircle, toAddCircle_intCast, Function.Periodic.lift_coe, expMapCircle_apply]
+  rw [toCircle, AddChar.coe_mk, AddCircle.toCircle, toAddCircle_intCast,
+    Function.Periodic.lift_coe, Circle.exp_apply]
   push_cast
   ring_nf
```

Comment:
May I ask why you are making this change? It does not seem to be necessary for the rest of the PR; and the new definition of `ZMod.toCircle` is much longer than the old one.